src/ansi-c/c_typecast.cpp

```diff
@@ -619,8 +619,10 @@ void c_typecastt::implicit_typecast_arithmetic(
   if(max_type==LARGE_SIGNED_INT || max_type==LARGE_UNSIGNED_INT)
   {
     // get the biggest width of both
-    std::size_t width1 = to_bitvector_type(type1).get_width();
-    std::size_t width2 = to_bitvector_type(type2).get_width();
+    std::size_t width1 =
```

Comment:
Would it make sense to use c_type1 and c_type2 instead of type1 and type2?

Reply:
Agreed; this should be post-promotion.

Reply:
But then we'd need new facilities to turn a `c_typet` into a `typet`. It's kind-of done by c_typecastt::implicit_typecast_arithmetic, but that doesn't do anything about a `LARGE_SIGNED_INT` (or `LARGE_UNSIGNED_INT`).

Reply:
I've now implemented this without a need for `to_bitvector_type` at all. Maybe I'm missing something? The new implementation seems almost too simple.